crates/bevy_ecs/src/system/system_param.rs

```diff
@@ -162,26 +164,22 @@ pub unsafe trait SystemParamState: Send + Sync + 'static {
 /// This must only be implemented for [`SystemParam`] impls that exclusively read the World passed in to [`SystemParamState::get_param`]
 pub unsafe trait ReadOnlySystemParam: SystemParam {}
 
-impl<'w, 's, Q: WorldQuery + 'static, F: ReadOnlyWorldQuery + 'static> SystemParam
-    for Query<'w, 's, Q, F>
-{
-    type State = QueryState<Q, F>;
-}
+pub type SystemParamItem<'w, 's, P> = <P as SystemParam>::Item<'w, 's>;
 
 // SAFETY: QueryState is constrained to read-only fetches, so it only reads World.
 unsafe impl<'w, 's, Q: ReadOnlyWorldQuery + 'static, F: ReadOnlyWorldQuery + 'static>
     ReadOnlySystemParam for Query<'w, 's, Q, F>
 {
 }
 
-// SAFETY: Relevant query ComponentId and ArchetypeComponentId access is applied to SystemMeta. If
-// this QueryState conflicts with any prior access, a panic will occur.
-unsafe impl<Q: WorldQuery + 'static, F: ReadOnlyWorldQuery + 'static> SystemParamState
-    for QueryState<Q, F>
+// SAFETY: QueryState is constrained to read-only fetches, so it only reads World.
```

Comment:
This safety comment doesn't make sense to me. Only F is constrained to be read only here, and I don't see how this lines up with the safety requirements on SystemParam.

Reply:
Probably a rebase error, thanks for catching it.

Reply:
Ah, this was copied from above by mistake! It's correct there, but totally wrong here.